### ISSUE TITLE
When upgrading to a new version of gradle, this Action will still request the old Gradle Wrapper folder cache.

### ISSUE THREAD

**Issue by VoidxHoshi** [gradle/gradle-build-action#388]
When upgrading to a new version of gradle, this Action will still request the old Gradle Wrapper folder cache. It's useless.

like
( gradle 7.5 to 7.5.1)

```
Entry: Gradle User Home
    Requested Key : v6-gradle|Linux|linux[37a6259cc0c1dae299a7866489dff0bd]-d9928975069c43547500d6692168eb04a2bbb3dc
    Restored  Key : v6-gradle|Linux|linux[37a6259cc0c1dae299a7866489dff0bd]-f4826ea727387f366d979df17de9e244564a40e5
              Size: 442 MB (463148968 B)
              (Entry restored: partial match found)
    Saved     Key : v6-gradle|Linux|linux[37a6259cc0c1dae299a7866489dff0bd]-d9928975069c43547500d6692168eb04a2bbb3dc
              Size: 442 MB (463591278 B)
              (Entry saved)
---
Entry: /home/runner/.gradle/wrapper/dists/gradle-7.5-bin/f1w0cc9nuo0eivz97x2xlu9sv
    Requested Key : wrapper-zips-d6e90d1ed428417836fce21db41bbf6b
    Restored  Key : wrapper-zips-d6e90d1ed428417836fce21db41bbf6b
              Size: 114 MB (119908388 B)
              (Entry restored: exact match found)
    Saved     Key : 
              Size: 
              (Entry not saved: contents unchanged)
---
Entry: /home/runner/.gradle/wrapper/dists/gradle-7.5.1-bin/7jzzequgds1hbszbhq3npc5ng
    Requested Key : wrapper-zips-a813b685d52d19a546025df7bcf77d62
    Restored  Key : wrapper-zips-a813b685d52d19a546025df7bcf77d62
              Size: 114 MB (119941789 B)
              (Entry restored: exact match found)
    Saved     Key : 
              Size: 
              (Entry not saved: contents unchanged)
---
Entry: /home/runner/.gradle/caches/modules-*/files-*/*/*/*/*
    Requested Key : dependencies-64e141e905dcc735476cc7da59e88e5a
    Restored  Key : dependencies-64e141e905dcc735476cc7da59e88e5a
              Size: 346 MB (362914659 B)
              (Entry restored: exact match found)
    Saved     Key : 
              Size: 
              (Entry not saved: contents unchanged)
---
Entry: /home/runner/.gradle/caches/jars-*/*
    Requested Key : instrumented-jars-64cf44b06ccc8e07c415f964990ddd37
    Restored  Key : instrumented-jars-64cf44b06ccc8e07c415f964990ddd37
              Size: 144 MB (150838946 B)
              (Entry restored: exact match found)
    Saved     Key : 
              Size: 
              (Entry not saved: contents unchanged)
```

**Comment by bigdaz** [comment]
You're correct: Gradle Wrapper distributions are stored in the Gradle User Home directory, and this action doesn't do anything special to cleanup unused files in the Gradle User Home before saving it to the GitHub Actions cache. 

Gradle doesn't currently provide any proper mechanism to force this cleanup, but I have implemented experimental support by setting various file timestamps in [this PR](https://github.com/gradle/gradle-build-action/pull/330). I hope to soon merge that PR as an opt-in feature.

**Comment by bigdaz** [comment]
Closing as superseded by #406

**Comment by bigdaz** [comment]
@VoidxHoshi Experimental support for Gradle User Home cleanup has been released in `v2.3.0-beta.1`. This should trigger removal of both old wrapper folders as well as unused local build cache entries.

Please try this version (with cache-cleanup enabled) and let us know how it works for you.

```
- uses: gradle/gradle-build-action@v2.3.0-beta.1
  with:
    gradle-home-cache-cleanup: true
```
